clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -402,26 +403,38 @@ public void refreshCommittedOffsetsIfNeeded() {
         }
     }
 
-    @Override
-    public void close() {
+    public void close(long timeoutMs, long requestTimeoutMs) {
         // we do not need to re-enable wakeups since we are closing already
         client.disableWakeups();
+        long startTimeMs = time.milliseconds();
         try {
-            maybeAutoCommitOffsetsSync();
+            timeoutMs = Math.min(timeoutMs, requestTimeoutMs);
+            maybeAutoCommitOffsetsSync(timeoutMs);
+        } finally {
+            super.close();
```

Comment:
Thanks, that makes sense. However, I think there may still be a problem. Any unsent requests will be targeted to the current coordinator at the time of `close()`. Discovering a new coordinator in the loop below won't help us unless we can retarget those unsent requests. What's more, we have logic in `AbstractCoordinator.coordinatorDead` to explicitly fail unsent requests to the coordinator.